packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -782,12 +900,32 @@ class MemoryBodyState extends State<MemoryBody>
         displayValue = name.substring(startOfNumber + 1);
       }
 
-      return [
-        image == null
+      Widget traceColor;
+      if (colorPatch != null) {
+        if (dashed) {
+          traceColor = Padding(
+            child: CustomPaint(
+              painter: DashedLine(15, colorPatch, 2, 4, 3),
+              foregroundPainter: DashedLine(15, colorPatch, 2, 4, 3),
+            ),
+            padding: const EdgeInsets.fromLTRB(2, 5, 20, 1),
+          );
+        } else {
+          traceColor = Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 1.0),
+            child: Container(height: 6, width: 20, color: colorPatch),
+          );
+        }
+      } else {
```

Comment:
this is duplicate code from above - consolidate into a helper if possible

Reply:
Yes it is, added two helpers.

```
  static const totalDashWidth = 15.0;
  static const dashHeight = 2.0;
  static const dashWidth = 4.0;
  static const spaceBetwenDash = 3.0;

  Widget createDashWidget(Color color) {
    return Container(
      padding: const EdgeInsets.only(right: 20),
      child: CustomPaint(
        painter: DashedLine(
          totalDashWidth,
          color,
          dashHeight,
          dashWidth,
          spaceBetwenDash,
        ),
        foregroundPainter: DashedLine(
          totalDashWidth,
          color,
          dashHeight,
          dashWidth,
          spaceBetwenDash,
        ),
      ),
    );
  }

  Widget createSolidLine(Color color) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 1.0),
      child: Container(
        height: 6,
        width: 20,
        color: color,
      ),
    );
  }
```

Now code:
```
      Widget traceColor;
      if (color != null) {
        if (dashedLine) {
          traceColor = createDashWidget(color);
        } else {
          traceColor = createSolidLine(color);
        }
      } else {
        traceColor =
            image == null ? const SizedBox() : Image(image: AssetImage(image));
      }
```